src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -431,21 +427,22 @@ public void kill(Map<String, String> modelEnvVars) throws IOException, Interrupt
 
                 int exitCode = doLaunch(
                         true, null, null, null, null,
+                        // TODO Windows
```

Comment:
Have not managed to come up with a PowerShell equivalent. `Get-Process` and `Stop-Process`, fine, but how do I select just those processes matching the `JENKINS_SERVER_COOKIE`? You would think [`Process.StartInfo`](https://docs.microsoft.com/dotnet/api/system.diagnostics.process.startinfo) would offer this via its `Environment` or `EnvironmentVariables` (what is the difference?), but this property is always null in my tests, perhaps because

> For example, you should not access the `StartInfo` property on a `Process` object returned by `GetProcesses`.

Reply:
Hmm, that seems to be true from an interactive shell, but then in other contexts this is set. Needs to be investigated further.

Reply:
Or rather, it prints `durable-XXX`…but for _every_ process, even for example `0 Idle` and `4 System`, so either the `StartInfo` is somehow poisoned by later state, or the `set` from `cmd` somehow mutated global state.

There is also the fact that `PowershellScript` overrides `doLaunch` when it should be overriding `launchWithCookie`, so even if we _could_ identify only processes actually started with this `JENKINS_SERVER_COOKIE`, we would wind up killing the controller process and getting a -1 return code and losing any final output.

So for now I think this is just unfixable.